### ISSUE TITLE
Create ledger stuck due to zk-connection loss

### ISSUE THREAD

**Issue by rdhabalia** [apache/incubator-pulsar#989]
#### Actual behavior

Broker creates a new ml-ledger when current ledger gets full and it doesn't publish any new messages until new ledger creation completes. However, [For some reason]( https://issues.apache.org/jira/browse/ZOOKEEPER-1277) if zookeeper quorum restarts and zk-connection is lost at broker then bk-client fails to create new ledger and it should fail broker's create-ledger callback to notify broker. But Bk-client doesn't complete the broker's callback so, related topic gets stuck and producer is not be able to produce any new messages until we unload the topic.
We saw following exception when broker lost the zk-connection
```
05:43:35.364 [main-EventThread] ERROR o.a.b.meta.ZkLedgerIdGenerator       - Could not generate new ledger id
org.apache.zookeeper.KeeperException$ConnectionLossException: KeeperErrorCode = ConnectionLoss for /ledgers/idgen-long/HOB-0000000001/ID-
        at org.apache.zookeeper.KeeperException.create(KeeperException.java:99) ~[zookeeper-3.4.6.jar:3.4.6-1569965]
        at org.apache.zookeeper.KeeperException.create(KeeperException.java:51) ~[zookeeper-3.4.6.jar:3.4.6-1569965]
        at org.apache.bookkeeper.meta.ZkLedgerIdGenerator$1.processResult(ZkLedgerIdGenerator.java:78) ~[bookkeeper-server-4.3.1.56-yahoo.jar:4.3.1.56-yahoo]
        at org.apache.bookkeeper.util.ZkUtils$1.processResult(ZkUtils.java:79) [bookkeeper-server-4.3.1.56-yahoo.jar:4.3.1.56-yahoo]
        at org.apache.bookkeeper.zookeeper.ZooKeeperClient$9$1.processResult(ZooKeeperClient.java:527) [bookkeeper-server-4.3.1.56-yahoo.jar:4.3.1.56-yahoo]
        at org.apache.zookeeper.ClientCnxn$EventThread.processEvent(ClientCnxn.java:605) [zookeeper-3.4.6.jar:3.4.6-1569965]
        at org.apache.zookeeper.ClientCnxn$EventThread.run(ClientCnxn.java:498) [zookeeper-3.4.6.jar:3.4.6-1569965]

```

Therefore, we can fix bk-client issue and upgrade the bk-version at broker.

Bk-client PR: https://github.com/yahoo/bookkeeper/pull/14

**Comment by rdhabalia** [comment]
@merlimat can you please review [yahoo/bookkeeper#14](https://github.com/yahoo/bookkeeper/pull/14) as we see zk-restart in multiple colo and many of the topics get stuck due to this issue.

**Comment by merlimat** [comment]
Nice catch!

**Comment by merlimat** [comment]
Fixed in #998
